src/org/parosproxy/paros/view/MainFrame.java

```diff
@@ -644,12 +646,34 @@ public void setShowTabNames(boolean showTabNames) {
 	/**
 	 * Sets the title of the main window.
 	 * <p>
+	 * The actual title set is the current session name, then the filename of the session (if persisted), 
+	 * followed by the program name and version.
+	 * 
+	 * @see Constant#PROGRAM_NAME
+	 * @see Constant#PROGRAM_VERSION
+	 * @since TODO add version
+	 */
+	public void setTitle() {
+		StringBuilder strBuilder = new StringBuilder();
+		strBuilder.append(Model.getSingleton().getSession().getSessionName()).append(" - ");
+		if (!Model.getSingleton().getSession().isNewState()) {
+			File file = new File(Model.getSingleton().getSession().getFileName());
+			strBuilder.append(" - ").append(file.getName().replaceAll(".session\\z", "")).append(" - ");
+		}
+		strBuilder.append(Constant.PROGRAM_NAME).append(' ').append(Constant.PROGRAM_VERSION);
+		super.setTitle(strBuilder.toString());
+	}
+	
+	/**
+	 * Sets the title of the main window.
+	 * <p>
 	 * The actual title set is the given {@code title} followed by the program name and version.
 	 * 
 	 * @see Constant#PROGRAM_NAME
 	 * @see Constant#PROGRAM_VERSION
+	 * @deprecated as of TODO add version
 	 */
-	@Override
+	@Override @Deprecated

```

Comment:
Should these be on separate lines?

Should I make the body of this function simply be a call to the new function?